DAOS-17259 test: Adding gcov to daos build

dinghwah opened the pull request:
Adding code-coverage report for daos unit test.

### Steps for the author:

* [ ] Commit message follows the [guidelines](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Appropriate [Features or Test-tag](https://daosio.atlassian.net/wiki/spaces/DC/pages/10984259629/Test+Tags) pragmas were used.
* [ ] Appropriate [Functional Test Stages](https://daosio.atlassian.net/wiki/spaces/DC/pages/12147556353/CI+Functional+Test+Stages) were run.
* [ ] At least two positive code reviews including at least one code owner from each category referenced in the PR.
* [ ] Testing is complete. If necessary, forced-landing label added and a reason added in a comment.

#### After all prior steps are complete:
* [ ] Gatekeeper requested (daos-gatekeeper added as a reviewer).

dinghwah committed:
DAOS-17259 test: Adding gcov to daos build

Adding code-coverage report for daos unit test.

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update daos_builder.py

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update _test_program

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update _program

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Add code_coverage with return kwargs

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update _static_library

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update _static_library env

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update with isinstenance

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
using _known_deps

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Restore and test

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update with env

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update _add_code_coverage

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update _static_library

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Restore _static_library

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update _add_code_coverage with env

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update _add_code_coverage

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update _add_code_coverage

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update debug on _add_code_coverage without env

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update _add_code_coverage with env

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update _add_code_coverage debug

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update _add_code_coverage for debug

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Restore _add_code_coverage

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update control/SConscript with gcov

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update _add_code_coverage debug

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update _add_code_coverage for debug

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
_add_code_coverage debug

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update _add_code_coverage debug

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update _add_code_coverage debug

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Restore _add_code_coverage

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update with gcov

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Remove env from _add_code_coverage

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Add debug for _run_command

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update control/SConscript

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Add debug _add_code_coverage

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update _add_code_coverage after debug

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update _program _test_program debug

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update _program _test_program debug

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Add env to _add_code_coverage

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Skip gcov for target daos_server_helper

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update control install_go_bin() for debug

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Restore SConscript from src/control

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update  CGO_LDFLAGS--coverage -g -O0

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update  CGO_LDFLAGS to _run_command

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update CGO_CFLAGS and code_coverage cleanup.

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update typo -lgcov

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update -fprofile-arcs

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update with debug

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Remove commented env.AppendUnique from _base_setup

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Sync with gcov unit-test and files

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest2

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest2

dinghwah committed:
Update with yaml

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Relocate test gcov env var

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

phender committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest2

phender committed:
Merge branch 'dinghwah/DAOS-17259-gtest2' of https://github.com/daos-stack/daos into dinghwah/DAOS-17259-gtest2

phender committed:
Fix pyyaml import.

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Add pyyaml install for code coverage

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

dinghwah committed:
Update junit_xml

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

jolivier23 commented on site_scons/site_tools/compiler_setup.py:
```diff
@@ -121,6 +121,11 @@ def _base_setup(env):
+    if GetOption('code_coverage'):
+        env.AppendUnique(CXXFLAGS=['-fprofile-arcs', '-ftest-coverage'])
+        env.AppendUnique(CCFLAGS=['-fprofile-arcs', '-ftest-coverage'])
+        env.AppendUnique(CGO_CFLAGS=['-fprofile-arcs', '-ftest-coverage'])
```
I don't think this has any effect.  It needs to be AppendENVPath('CGO_CFLAGS', ...) similar to CGO_LDFLAGS below

dinghwah replied:
Jeff, Thanks for the review and comments.
        I comment-out above lines and started a scons-3 build, however observe following error:
scons: *** [build/release/gcc/src/control/daos] CalledProcessError : Command '['/usr/bin/go', 'list', '--json', '-mod=vendor', 'cmd/daos/acl.go']' returned non-zero exit status 1.
scons: building terminated because of errors.

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/15/testReport/

dinghwah committed:
Update with python3.11 and debug

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Recommit by adding a debug comment

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Debug with ls lib64

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Comment out the cp

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Add setup.py and update cp storage_estimator.

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Add debug  ls -la storage_estimator

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/25/testReport/

dinghwah committed:
Update debug cp venv/lib64

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/26/testReport/

dinghwah committed:
Update cp libvos_size.so to venv

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update cp daos_srv/libvos_size.so to daos_srv

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage NLT on EL 8.8 completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos/job/PR-16338/27/display/redirect

dinghwah committed:
Update cp -r daos_srv.

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
cp libdfs.so to lib64

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update to increase timeout

Skip-nlt: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest2

Skip-nlt: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/32/testReport/

dinghwah committed:
Update Unit Test bdev with memcheck timeout to 360

Skip-nlt: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/33/testReport/

dinghwah committed:
Empty commit L#857 Jenkinsfile

Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/34/testReport/

dinghwah committed:
Update with ./requirements-code-coverage.txt

Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/36/testReport/

dinghwah committed:
Update ARG BASE_DISTRO.

Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true
Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Recommit

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/37/testReport/

dinghwah committed:
Update Unit Test bdev with memcheck to 480

Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update COPY requirements-code-coverage.txt /tmp

Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/41/testReport/

dinghwah committed:
Add debug pwd

Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/42/testReport/

dinghwah committed:
Update with pip install

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update pip install gcovr

Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true
Allow-unstable-test: true
Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update with COPY .txt to ./

Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true
Allow-unstable-test: true
Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/44/testReport/

dinghwah committed:
Update to skip COPY .txt /tmp/

Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true
Allow-unstable-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/45/testReport/

daosbuild3 commented:
Test stage Test RPMs on EL 8.6 completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos/job/PR-16338/45/display/redirect

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest2

Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true
Allow-unstable-test: true
Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/46/testReport/

daosbuild3 commented:
Test stage Test RPMs on EL 8.6 completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos/job/PR-16338/46/display/redirect

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/47/testReport/

dinghwah committed:
Update COPY ./requirements-code-coverage.txt ./

Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true
Allow-unstable-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/48/testReport/

dinghwah committed:
Update Unit Test bdev with memcheck to 720 for debug

Skip-nlt: true
Allow-unstable-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/49/testReport/

phender committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest2

phender committed:
Updated dockerfile for code coverage report.

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/51/testReport/

dinghwah committed:
Recommit Phil's pr#51 with UT_memcheck

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-nlt: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/61/testReport/

dependabot[bot] committed:
DAOS-17664 cq: update gha versions (#16490)

Updates `github/codeql-action` from 3.28.18 to 3.28.19
Updates `aquasecurity/trivy-action` from 0.30.0 to 0.31.0

Signed-off-by: dependabot[bot] <support@github.com>

phender commented on utils/docker/Dockerfile.test_summary:
```diff
@@ -0,0 +1,57 @@
+RUN python3 -m venv /home/daos/venv
+ENV PATH=/home/daos/venv/bin:$PATH
+ENV VIRTUAL_ENV=/home/daos/venv/
+
```
```suggestion

WORKDIR /home/daos

```

dinghwah committed:
Add WORKDIR before COPY

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

kjacque committed:
DAOS-17432 test: Update suppressions for Go runtime (#16310)

Based on NLT Valgrind results, it seems the path for internal
Go runtime functions has changed with the latest Go version.

Signed-off-by: Kris Jacque <kris.jacque@hpe.com>

shimizukko committed:
DAOS-17651 test: Increase timeout for recovery/pool_list_consolidatio… (#16485)

test_lost_majority_ps_replicas timed out at pool destroy
during tearDown (while other tests passed). Increase
the timeout by 45 sec.

Signed-off-by: Makito Kano <makito.kano@hpe.com>

dinghwah committed:
Update .dockerignore with requirements-code-coverage.txt

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/65/testReport/

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/70/testReport/

phender committed:
Make ci/code_coverage_report.sh executable.

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

dinghwah committed:
Update with find jsonfiles for gcovr.

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/72/testReport/

dinghwah committed:
Add double quote for jsonfiles

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update with --add-traceback

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update with --add-tracefile $jsonfiles

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/75/testReport/

dinghwah committed:
Add test only --add-tracefile for debug

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/76/testReport/

dependabot[bot] committed:
DAOS-17705 cq: Bump github/codeql-action from 3.28.19 to 3.29.0 (#16513)

Updates `github/codeql-action` from 3.28.19 to 3.29.0

Signed-off-by: dependabot[bot] <support@github.com>

dinghwah committed:
Try python3.11

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/77/testReport/

dinghwah committed:
Update RUN apt-get -qqy install python3.11

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/78/testReport/

phender committed:
Use dnf

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

kjacque committed:
DAOS-17049 control: Allow graceful shutdown for specific ranks (#16305)

The CR checker tool needs to use a graceful shutdown when stopping
ranks. It may select a subset of ranks if some are admin-excluded.

- Remove the limitation that non-forced shutdown may only be used
  on the whole system, not while specifying ranks.

Signed-off-by: Kris Jacque <kris.jacque@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/79/testReport/

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/80/testReport/

frostedcmos committed:
DAOS-17594 cart: allocate iov buffer for deferred bulk #16430 (#16454)

Allocate iov buffer for deferred bulk, since input sgl
may use temporary iovs, which may cause memory corruption
or data corruption if they are used by the following
bulk creation.

Signed-off-by: Di Wang <ddiwang@google.com>
Signed-off-by: Alexander A Oganezov <alexander.oganezov@hpe.com>

dinghwah committed:
Remove export GCOV_PREFIX

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest2

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update GCOV_PREFIX_STRIP=7

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Tmp comment out mv nlt-errors.json

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Restore test_nlt_post and update $pwd)

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update rsync to /tmp/code_coverage

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update include code_coverage/*

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-func-test: true
Allow-unstable-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Add debug /tmp/code_coverage after nlt gcovr

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update rsync .. include code_coverage*

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Remove the unneed comment statements

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Allow-unstable-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/91/testReport/

dinghwah committed:
Cleanup and debug for required file.

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true
Allow-unstable-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/93/testReport/

dinghwah committed:
Debug commented out CGO_CFLAGS per Jeff's comment

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-func-test: true
Allow-unstable-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Tmp comment out env.AppendENVPath('CGO_LDFLAGS'..) per Jeff's comment

Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-func-test: true
Allow-unstable-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16338/95/execution/node/313/log

dinghwah committed:
Restore env.AppendENVPath('CGO_LDFLAGS'..)

Skip-func-test: true
Allow-unstable-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16338/95/execution/node/310/log

daosbuild3 commented:
Test stage Build RPM on Leap 15.5 completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16338/95/execution/node/287/log

dinghwah committed:
Update /daos_srv

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-unit-test-memcheck: true
Skip-nlt: true
Skip-func-test: true
Allow-unstable-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Cleanup and all unit-tests

Skip-func-test: true
Allow-unstable-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Restore env.AppendUnique(CGO_CFLAGS=..)

Skip-func-test: true
Allow-unstable-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update env.AppendENVPath('CGO_LDFLAGS',..)

Skip-func-test: true
Allow-unstable-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Restore env.AppendUnique(CGO_CFLAGS=..)

Skip-func-test: true
Allow-unstable-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update for time build

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Fixed typo for startedByTimer

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Fix skip logic

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/105/testReport/

dinghwah committed:
Test for python3

Skip-func-test: true
Allow-unstable-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Restore back to python3.11

Skip-func-test: true
Allow-unstable-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest2

Skip-func-test: true
Allow-unstable-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Tmp comment out build on EL9 and Leap15.5 stages

Skip-func-test: true
Allow-unstable-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update is_code_coverage to Build on EL9 and Leap15.5

Skip-func-test: true
Allow-unstable-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest2

Skip-func-test: true
RPM-test: false
Allow-unstable-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Test RPMs on EL 8.6 completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos/job/PR-16338/111/display/redirect

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest2

Skip-func-test: true
Allow-unstable-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest2

Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Allow-unstable-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest2

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Add debug display *.gcno & *.gcda

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16338/115/execution/node/474/log

daosbuild3 commented:
Test stage Unit Test bdev on EL 8.8 completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16338/115/execution/node/455/log

dinghwah committed:
Added gcovr gcov-ignore-error

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest2

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

phender committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest2

phender committed:
pipeline-lib updates.

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest2

phender committed:
Go back to pipeline-lib hendersp/DAOS-17259

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Fix typo.

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/127/testReport/

daosbuild3 commented:
Test stage NLT on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/126/testReport/

phender committed:
Installing python 3.11 in el8 docker image

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Revert previous change.

Install python3.11 in code_coverage_report.sh.

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Use sudo dnf

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/133/testReport/

phender committed:
Update python

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Adding env.PYTHON_VERSION.

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Hardcode python version.

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Trying a different approach

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Fix pip install.

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Move to python 3.11

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Usse same python version installed in dockerfile

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Adding -vvv to pip

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest2

phender committed:
Trying separate dockerfile for summary stage.

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Removing no longer used --build-arg.

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Fix previous edit.

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest2

phender committed:
Updated requirements files.

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Trying setting pip proxy.

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Use the correct python version in the dockerfile

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Add missing Dockefile ARG.

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Remove the no longer needed storage_estimator workaround

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/165/testReport/

phender committed:
Fix pip config setup.

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Adding pip debug.

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Fix pip proxy setting.

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/171/testReport/

daosbuild3 commented:
Test stage NLT on EL 8.8 completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16338/172/execution/node/690/log

phender committed:
Updates.

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/174/testReport/

phender commented on ci/unit/test_nlt_node.sh:
```diff
@@ -25,21 +25,32 @@ sudo bash -c ". ./utils/sl/setup_local.sh; ./utils/setup_daos_server_helper.sh"
 prlimit -n
 
+cwd="$(pwd)"
+export GCOV_PREFIX="${cwd}"
+export GCOV_PREFIX_STRIP="7"
+
 HTTPS_PROXY="${DAOS_HTTPS_PROXY:-}" ./utils/node_local_test.py --max-log-size 1950MiB \
```
```suggestion

GCOV_PREFIX="${cwd}" GCOV_PREFIX_STRIP="7" HTTPS_PROXY="${DAOS_HTTPS_PROXY:-}" ./utils/node_local_test.py --max-log-size 1950MiB \
```

dinghwah committed:
Update HTTPS_PROXY GCOV_PREFIX

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/176/testReport/

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest2

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Fix spaces.

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/179/testReport/

daosbuild3 commented:
Test stage NLT on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/179/testReport/

daosbuild3 commented:
Test stage Functional Hardware Medium MD on SSD completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/178/testReport/

dinghwah committed:
Update 	CGO_LDFLAGS with lgcov

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/182/testReport/

dinghwah committed:
Update find \*.gcno

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/184/testReport/

dinghwah committed:
Update find src

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Functional Hardware Medium MD on SSD completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/185/testReport/

dinghwah committed:
Update path to find gcno file

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update find

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/190/testReport/

phender committed:
Fix run_go_tests CGO_LDFLAGS assignment for code coverage.

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Skip running abt_stack with code coverage.

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Fix code coverage detection for unit tests

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

daosbuild3 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/196/testReport/

phender committed:
Fix skipping abt_stack

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Fix typo and exclude NLT w/ code coverage

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest2

phender committed:
Moving requirements-*.txt changes to PR-16919.

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

daosbuild3 commented:
Test stage NLT on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/204/testReport/

daosbuild3 commented:
Test stage Functional Hardware Medium MD on SSD completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16338/204/execution/node/1160/log

phender committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest2

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest2

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest2

phender committed:
Fix GHA ubuntu python version.

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Fix ubuntu python version

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender commented:
> Summary of updated test results:
> 
> * [Build 206](https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos/job/PR-16338/206) (w/o CI_CODE_COVERAGE checked) - failed Fault Injection
> * [Build 207](https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos/job/PR-16338/207) (w/o CI_CODE_COVERAGE checked) - re-run of 206.  Passed Fault-Injection.  One functional test failure - known issue [DAOS-18218](https://daosio.atlassian.net/browse/DAOS-18218)
> * [Build 208](https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos/job/PR-16338/208) (w/ CI_CODE_COVERAGE checked) - passed w/ [code coverage report](https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos/job/PR-16338/208/artifact/code_coverage_report/code_coverage_report.html).

[Build 210](https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos/job/PR-16338/210) (w/o CI_CODE_COVERAGE) passed

ryon-jensen commented:
What's the expected impact to CI of the code coverage stage? If I understand the Jenkins file changes correctly, the stage will only run once a week, and only for unit tests? Is it a new scheduled run, or extra stage for an already existing scheduled run? 
Also, the increase in timeouts applies to all builds, not just when code coverage is enabled, correct?

phender commented on utils/docker/Dockerfile.ubuntu:
```diff
@@ -64,14 +64,16 @@ RUN mkdir -p /opt/daos /mnt/daos /var/run/daos_server /var/run/daos_agent /home/
 COPY requirements-build.txt requirements-utest.txt ./
-RUN python3 -m pip --no-cache-dir install --upgrade pip && \
-    python3 -m pip --no-cache-dir install -r requirements-build.txt -r requirements-utest.txt
+RUN source /home/daos/venv/bin/activate && \
```
```suggestion
RUN . /home/daos/venv/bin/activate && \
```

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest2

dinghwah committed:
Update per Phil's comment.

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Fix pylint issue

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild3 commented:
Test stage NLT on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16338/213/testReport/

dinghwah committed:
Update debug

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Add Debug to find build

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

phender committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest2

phender committed:
Debug

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

daltonbohning reviewed (commented):
@phender @dinghwah Do we still need this?
